src/server/main.js

```diff
@@ -63,9 +70,25 @@ io.on('connect', (socket) => {
    returns JWT
    */
   socket.on('login', async (user) => {
-    const login = await authentication.login(user);
+    const token = await authentication.login(user);
+    const tokenVerification = authentication.verifyToken(token);
+
+    if (tokenVerification !== false) {
+      socketUser = authentication.verifyToken(socket.handshake.query.token);
+    }
+    socket.emit('login', token);
+  });
 
-    socket.emit('login', login);
+  socket.on('uploadMemeData', async (data) => {
+    if (socketUser === false) {
+      return socket.emit('uploadMemeData', 'cannot verify user');
+    }
+    const saveResult = await meme.saveMeme(data, socketUser);
+
+    if (saveResult !== true) {
+      return socket.emit('uploadMemeData', saveResult);
+    }
+    return socket.emit('uploadMemeData', 'successfully saved meme');
```

Comment:
On the "happy path" return something the client can more reliably test like `true`. So you could probably remove the `if` block and last line, and go with the line inside the `if`